blogposts/2023-02-10-adding-snapshot-testing-to-rnw-apps.md

```diff
@@ -0,0 +1,93 @@
+
+test('Button Example Page', () => {
+  const tree = create(<ButtonExamplePage />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
+```
+
+As I began to work on this, I realized there was some additional configuration I needed to add in order to get my tests to run correctly. First, I had several modules that needed to be added to the `transformIgnorePatterns` Jest customization. This customization is used to specify which files should be transformed by Babel. By default, Jest only processes the app's and react-native's source code. You'll know if a module needs to be transformed, if the app throws an error saying "Jest encountered an unexpected token". In this case, the module should be listed in the `transformIgnorePatterns` customization within your app's `package.json`. See Jest's [transformIgnorePatterns](https://jestjs.io/docs/tutorial-react-native#transformignorepatterns-customization) documentation for syntax info.
+
+Next, I had several modules which required [mocking](https://jestjs.io/docs/mock-function-api/). Mock functions are often needed for modules which implement API's instead of UI components. Examples within React Native Gallery include the `react-native-device-info` module or the `react-navigation` module. If a function needs to be mocked by Jest, you'll commonly receive an error that the method is undefined, when trying to render a component which makes use of the method. Here's an example error for a `react-native-device-info` method that needs mocking: "`@react-native-community/react-native-device-info`: `NativeModule.RNDeviceInfo` is null". To add mocks to your Jest configuration, you'll want to create a `jest-setup.js` file at the root of your repo. Then you'll want to edit your Jest configuration in your app's `package.json` to identify that you've created a setup file (*see the `package.json` source snippet below for syntax*). Now you'll want to add contents to your `jest-setup.js` file. Some community modules already have `mock.js` files implemented in their source. If this is the case, you'll just want to reference this file in your setup file.
+
+Here is an example of the code added for the `react-native-permissions` module:
+```js
+jest.mock('react-native-permissions', () =>
+require('react-native-permissions/mock'),
+);
+```
+If the community module does not have a mock.js file, you'll have to mock the undefined function yourself. There are a couple ways to sleuth out how to correctly mock: 
+	1. Check the community module source. If they have an example app with Jest testing, what do they do in their `jest-setup.js` file?
+	2. You can also check forum posts from the community on how they mocked particular methods for a given module.
+	3. React Native Gallery has a wide set of community modules it has tests for. Check out our [`jest-setup.js`](https://github.com/microsoft/react-native-gallery/blob/main/jest-setup.js) to see how we mocked needed modules. 
```

Comment:
headsup, this section doesn't render correctly as a markdown numbered list; I think it might have to do with formatting/the tabs you added, you might also need a blank row between the phrase and 1. ([check the rendered file](https://github.com/microsoft/react-native-devblog/blob/chiaramooney-AddingSnapshotTestingToRNWApps/blogposts/2023-02-10-adding-snapshot-testing-to-rnw-apps.md))

Reply:
Will change!